conduit/src/Data/Conduit/Internal/Conduit.hs

```diff
@@ -720,6 +723,38 @@ connect :: Monad m
         -> m r
 connect = ($$)
 
+-- | Split a conduit into head and tail.
+--
+-- Note that you have to 'sealConduitT' it first.
+--
+-- Since 1.3.3
+uncons :: forall m o conduit. (Monad m, conduit ~ SealedConduitT () o m ())
+       => conduit -> m (Maybe (o, conduit))
+uncons (SealedConduitT p) = fmap (fmap (second SealedConduitT)) $ go p
+  where
+    -- This function is the same as @Pipe.uncons@ but it ignores leftovers.
+    go (HaveOutput p o) = pure $ Just (o, p)
```

Comment:
If I'm reading this correctly, instead of the `fmap` and `second` business above, you could apply `SealedConduitT` to `p` on this line. Personally I'd find that easier to read. Is there a reason to avoid that?